azure-pipelines.yml

```diff
@@ -285,7 +284,7 @@ stages:
         numpy==1.21.4
         Pillow
         pybind11
-        pythran
+        pythran==0.10.0
```

Comment:
We were already passing CI with Windows with Pythran `0.11`. I dissected the pass/fail build log comparisons and the first Pythran source file to fail to build (`_rbfinterp_pythran`) has only a single, but pretty major, difference in compile lines before vs. after, it is using a different compiler (was passing with: `clang-cl` driver; failing with plain `cl.exe` now):

```diff
diff --git a/windows_64bit_passing.txt b/windows_64bit_failing.txt
index 5f3d9c2..beb5bdf 100755
--- a/windows_64bit_passing.txt
+++ b/windows_64bit_failing.txt
@@ -1 +1 @@
[-clang-cl.exe-]{+C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\cl.exe+} /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_NONE -IC:\hostedtoolcache\windows\Python\3.9.9\x64\lib\site-packages/pythran -IC:\hostedtoolcache\windows\Python\3.9.9\x64\lib\site-packages\numpy\core\include -IC:\hostedtoolcache\windows\Python\3.9.9\x64\lib\site-packages\numpy\core\include -Ibuild\src.win-amd64-3.9\numpy\distutils\include -IC:\hostedtoolcache\windows\Python\3.9.9\x64\include -IC:\hostedtoolcache\windows\Python\3.9.9\x64\include -IC:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\include -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\cppwinrt -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\winrt /EHsc /Tpscipy/interpolate/_rbfinterp_pythran.cpp /Fobuild\temp.win-amd64-3.9\Release\scipy/interpolate/_rbfinterp_pythran.obj /std:c++14 /w /Zm1000
```

I'll see if I can get this passing today, perhaps via hard-specification of the CXX compiler in the env for now, but will try to take a little "Holiday time" though so I may give up :)

Reply:
Good catch. Any guesses as to what changed the compiler detection?

Reply:
The most recent pass for the failing Windows job I see was 6 days ago: https://github.com/scipy/scipy/pull/15244

Five days ago, it looks like the Azure/Actions base image may have changed: https://github.com/actions/virtual-environments/pull/4776 (looks like a visual studio version change in there to me?)

The 64-bit Windows failing job here shows: `Current image version: '20211219.1'`
The passing PR I linked above shows: `Current image version: '20211212.1'`

I don't actually see us taking much action to force the use of `clang-cl` in CI there in any case though (we seem to install it, but it is already there anyway.. but don't really change paths/env variables), and our Windows docs suggest fairly broad support including native MSVC (can Pythran say the same?). At https://pythran.readthedocs.io/en/latest/ I see: 

>Windows support is on going and only targets Python 3.5+ with either Visual Studio 2017 or, better, clang-cl:
>
> $> pip install pythran
> 
> Note that using clang-cl.exe is the default setting. It can be changed through the CXX and CC environment variables.

I made a crude attempt to do just that but not working yet: https://github.com/tylerjereddy/scipy/pull/59 

Maybe I need to add that numpy-distutils "APPEND" env variable as well or something? I thought that was just for compile flags rather than compilers proper though. Perhaps the absolute paths I used are wrong.. but seems like the best path I can think of given current info.

If Pythran is driving the `clang-cl` default selection that may be something to double check re: newer msvc perhaps, dunno.

Reply:
> and our Windows docs suggest fairly broad support including native MSVC (can Pythran say the same?)

It needs `clang-cl`. Note that we soon plan to change to a full Mingw-w64 toolchain anyway with Meson.





> If Pythran is driving the `clang-cl` default selection that may be something to double check re: newer msvc perhaps, dunno.

Yep, that's probably it.

Reply:
> Maybe I need to add that numpy-distutils "APPEND" env variable as well or something?

That's no longer a thing, it became irrelevant after numpy 1.18